### PR TITLE
Add N3LO ren SV

### LINKED ISSUE #144: Generated renormalization SV does not work at N3LO (and beyond)
[`ren_sv_coeffs`](https://github.com/NNPDF/pineko/blob/e2cb2415ff4c1a952956a5386dbee129b66217b5/src/pineko/scale_variations.py#L33) does not work at N3LO, as it seems hardcoded to NNLO:
https://github.com/NNPDF/pineko/blob/e2cb2415ff4c1a952956a5386dbee129b66217b5/src/pineko/scale_variations.py#L56-L63
If you don't want to do the fully general math (which I guess should be [Bell polynomials](https://en.wikipedia.org/wiki/Bell_polynomials) - look at the second formula image under properties), you could still do an explicit spell out (as N4LO is far future, hopefully) ...

The relevant coeffs can be found in the MHOU paper:
grafik

cc @giacomomagni

### PR DESCRIPTION
**Pull request by felixhekhorn**
Closes #144

I hope I did the correct thing :see_no_evil: ... (unfortunately the module is not so clearly written ...) if you agree I can add a N3LO unit test

### CODE REVIEW

**Review by andreab1997 — COMMENTED** [review]
Thanks for this. It looks ok (even if I need to have a more careful look). My only comment is that this function is only called to get one contribution, while in this way every time it needs to compute all the terms, while returning only one. This seems a waste to me.

**Comment by felixhekhorn** [comment]
> Thanks for this. It looks ok (even if I need to have a more careful look). My only comment is that this function is only called to get one contribution, while in this way every time it needs to compute all the terms, while returning only one. This seems a waste to me.

True - but I'm sure we can afford the calculation of O(10) polynomials for the sake of clarity

**Review by andreab1997 — APPROVED** [review]

File: src/pineko/scale_variations.py [hunk 1]
```diff
@@ -51,16 +51,32 @@ def ren_sv_coeffs(m, max_as, logpart, which_part, nf):
...
+        (2, 1, 1): (m + 1) * beta0,
+        (2, 1, 0): (m + 0) * beta1,
+        (2, 2, 0): m * (m + 1) / 2.0 * beta0**2,
+        # N3LO
+        (3, 1, 2): (m + 2) * beta0,
+        (3, 1, 1): (m + 1) * beta1,
+        (3, 1, 0): (m + 0) * beta2,
+        (3, 2, 1): (m + 1) * (m + 2) / 2.0 * beta0**2,
+        (3, 2, 0): m * (2 * m + 3) / 2.0 * beta0 * beta1,
+        (3, 3, 0): m * (m + 1) * (m + 2) / 6 * beta0**2,
```
Comment (giacomomagni) [inline src/pineko/scale_variations.py:77]:
```suggestion
        (3, 3, 0): m * (m + 1) * (m + 2) / 6 * beta0**3,
```

**Review by giacomomagni — CHANGES_REQUESTED** [review]

File: src/pineko/scale_variations.py (hunk 1 above)
Comment (andreab1997) [reply src/pineko/scale_variations.py:77]: Well spotted! Sorry I did not notice

Comment (felixhekhorn) [reply src/pineko/scale_variations.py]: Conclusion: triple check is not enough, always ask someone else :innocent:

**Review by giacomomagni — APPROVED** [review]

**Pull request merged**